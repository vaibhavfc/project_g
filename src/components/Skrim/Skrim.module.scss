@import '../../sass/theme/default';

.root {
  height: 35.5rem;
  width: 20rem;
}

.modal1 {
  background-color: rgba($skrim--model1, 0.24);
}

.modal2 {
  background-color: rgba($skrim--model2, 0.3);
}

.nonModal {
  background-color: rgba($skrim--nonmodel, 0.15);
}
