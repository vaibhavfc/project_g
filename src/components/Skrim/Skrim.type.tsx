// import { Ref } from 'react';
import { TypographyProps as MuiTypographyProps, TypographyClassKey } from '@material-ui/core/Typography';

export interface SkrimStylingProps extends Partial<Record<TypographyClassKey, string>> {
}

export interface SkrimProps extends MuiTypographyProps {
  // innerRef: Ref<HTMLDivElement>;
  type: 'modal1' | 'modal2' | 'nonModal';
  children ?: any;
}
