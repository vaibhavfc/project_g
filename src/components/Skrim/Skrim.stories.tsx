// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import Skrim from '.';

import classes from './SkrimWithClass.module.scss';

import { SkrimProps } from './Skrim.type';

export default {
  title: 'Components/Skrim',
  component: Skrim,
  parameters: {
    controls: {
      // disabled: true,
    },
  },
} as Meta;

const Template: Story<SkrimProps> = (args) => <Skrim {...args} />;

export const Modal1 = Template.bind({});
Modal1.args = {
  type: 'modal1',
};

export const Modal2 = Template.bind({});
Modal2.args = {
  type: 'modal2',
};

export const NonModal = Template.bind({});
NonModal.args = {
  type: 'nonModal',
};
