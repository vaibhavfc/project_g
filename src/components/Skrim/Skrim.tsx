import React, { FC, forwardRef, useMemo } from 'react';
import { Container as MuiContainer } from '@material-ui/core';
import classnames from 'classnames';
import { SkrimProps, SkrimStylingProps } from './Skrim.type';

import classes from './Skrim.module.scss';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const SkrimWithInnerRef: FC<SkrimProps> = ({
  // innerRef = null,
  classes: overrideClasses = {},
  // children = 'Skrim',
  type,
  // ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<SkrimStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );
  return (
    <MuiContainer
      classes={{
        root: classnames(
          mergedClasses.root, {
            [classes.modal1]: (type === 'modal1'),
            [classes.modal2]: (type === 'modal2'),
            [classes.nonModal]: (type === 'nonModal'),
          },
        ),
      }}
    />
  );
};

const Skrim = forwardRef<HTMLDivElement, Omit<SkrimProps, 'innerRef'>>(
  (props, ref) => <SkrimWithInnerRef innerRef={ref} {...props} />,
);
export default Skrim;
