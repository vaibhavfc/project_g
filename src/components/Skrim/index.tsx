import Skrim from './Skrim';
import { SkrimProps } from './Skrim.type';

export type { SkrimProps };

export default Skrim;
