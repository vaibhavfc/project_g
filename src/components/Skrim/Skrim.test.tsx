import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import Skrim from './Skrim';

describe('test', () => {
  it('Renders Skrim Compoents', () => {
    render(<Skrim>test</Skrim>);
  });

  it('Matches Snapshot of Skrim', () => {
    const tree = renderer.create(<Skrim> TEST</Skrim>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
