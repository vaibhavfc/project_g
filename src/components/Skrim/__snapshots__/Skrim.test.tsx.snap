// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of Skrim 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
/>
`;
